### ISSUE TITLE
Automated sync via Cron failing due to call to non-existent method

### ISSUE THREAD

**Issue by aj-adl** [facebookincubator/facebook-for-woocommerce#975]
We get regular failures when using the 'automated sync' option for the facebook plugin.

When referencing the error log we get the following error 

```2019-04-29T23:06:18+00:00 CRITICAL Uncaught Error: Call to undefined method WC_Facebookcommerce_Integration::sync_all_products() in /srv/users/xxxxxxxx/apps/path-to-wordpress/content/plugins/facebook-for-woocommerce/facebook-commerce.php:1862```

There are multiple issues that have been robo closed that reference this EXACT error, and it's very plain to see why this is an issue.

The [function called here ](https://github.com/facebookincubator/facebook-for-woocommerce/blob/master/facebook-commerce.php#L1862) simply does not exist. This is not an issue for all installs, as if the upload / sync via the feed succeeds then this statement is never run.

Why is this code calling a function that doesn't exist? It seems to be a problem for plenty of people as evident by the automated bug reports on here, but as those people aren't github users they are being robo-closed and this issue persists.

Can a human please look at this?

**Comment by aj-adl** [comment]
Debug Ready

**Comment by jieminz22** [comment]
## Automated Support Message

Hello,

Thanks for your interest in our plugin! Our developer team tries to provide personal support on GitHub whenever possible.

If you have a feature request or feedback for the developers, reply with `Feature Request` or `Feedback`.
If you are a developer or have a question for the developers related to code, reply with `Developer Question`.
If you are running ads with us, reply with `Advertiser`.

If you have encountered a bug or a problem with the plugin, keep reading for information that can help you resolve the issue yourself or provide us details we need to help you.

## Search Existing Issues

First, please [search our issues](https://github.com/facebookincubator/facebook-for-woocommerce/issues?q=) to see if your problem has been reported before.
There may already be a workaround or a solution to your problem. You can use the [search bar](https://github.com/facebookincubator/facebook-for-woocommerce/issues?q=) near the top of our issues page. Be sure to search both _open_ and _closed_ issues.
If you do find a similar issue, leave a comment on that issue instead of creating a new one. Your comment helps us prioritize which existing issues are important and urgent.

## Common Resolution Methods

If you are experiencing a problem or bug and have not found any similar issue, please try the following:

* Upgrade to the latest version of the plugin
* Try to `Resync` products using the button on the plugin config page
* Reset the plugin by selecting `Manage Settings > Advanced Options > Delete` from the configuration page of the plugin and setting up again
* Try deactivating other plugins that you think may be interfering, then reset and set up again

## Necessary Information

If the above steps did not help, please provide the following information in your next comment in order for us to debug your problem:

* WooCommerce version
* WordPress version
* Plugin version (please note: we only support up-to-date versions of the plugin)
* A screenshot of where you see the problem (for example: if it is on Facebook, provide a screenshot of Facebook, if it is on WooCommerce, provide a screenshot of WooCommerce. You can provide multiple screenshots if the problem relates to different screens.)
* The URL of your website
* The URL of your Facebook Page
* The URL(s) of where you took your screenshot(s)
* If applicable, detailed steps you took in order to see the problem

Please include `Debug Ready` in your reply so our automated system can note your issue for manual review. Please be aware that due to a high volume of issues, we may not be able to respond immediately.

## Response Time

We prioritize issues with the tags `Advertiser` or `Multiple Reports`.
For issues that have the tags `Debug Ready`, `Feature Request` or `Developer Question`, we aim to respond within 30 days.

If you have not responded to our automated message with one of the above phrases, such as `Debug Ready`, we may close your issue after 30 days. If a `Feature Request` is inactive (without new requests, messages or pull requests) for 6 months, we may close it.

Please note that we are not always able to provide support for issues related to interaction from other plugins with this plugin or related to the Facebook advertising interface. To get support related to Facebook ads, try visiting the [Facebook Ads Help Center](https://www.facebook.com/business/help).

For more information on our response times, visit [Facebook Support on GitHub](https://github.com/facebookincubator/facebook-for-woocommerce/wiki/Facebook-Support-on-GitHub).

Thank you,
The Facebook Dev Team

**Comment by aj-adl** [comment]
Debug Ready

Affects - all  recent versions, woo version doesn't matter. Can a human please look at this, instead of this robot closing the tickets automatically. Any test site will work. You do not need our specific implementation, you only need to read the code for 2 minutes to understand it's IMPOSSIBLE for this to work.

Related Tickets (that have mostly been robo-closed due to the issuer not responding (not via ACTUAL resolution).

**Github**
859
779
908
901

In all cases where this was 'resolved' or working, it's because the sync via catalog worked, the underlying issue that if this sync fails it calls a non-existen function, has been present since at least NOVEMBER 2018.

**WordPress.org related cases**
https://wordpress.org/support/topic/sync-feature-error-500/

**Comment by jieminz22** [comment]
Hi @aj-adl,

Thanks for reporting this again. We are working on a fix and will release a new version soon. Meanwhile if you want to unblock, you can remove that else block to get rid of the exception. Let us know if it doesn't work.

**Comment by lgfogle** [comment]
Hey there! 👋

Lindsey here, from the SkyVerge product team! We recently took over development and support for the Facebook for WooCommerce extension.

In our recent 1.10 releases, we've fixed many issues impacting the store connection, pixel tracking, and product sync. Could you please update to the latest version of Facebook for WooCommerce and see if that resolves this issue?

If not, could you please [contact our support team](https://woocommerce.com/my-account/create-a-ticket/) or open a new issue here? We'd be happy to help troubleshoot further with you.

Thank you!
